src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/CodeActions/RazorCodeActionRunner.ts

```diff
@@ -0,0 +1,34 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+import { RazorLanguageServerClient } from '../RazorLanguageServerClient';
+import { RazorLogger } from '../RazorLogger';
+import { CodeActionResolutionResponse } from '../RPC/CodeActionResolutionResponse';
+import { convertWorkspaceEditFromSerializable } from '../RPC/SerializableWorkspaceEdit';
+
+export class RazorCodeActionRunner {
+
+    constructor(
+        private readonly serverClient: RazorLanguageServerClient,
+        private readonly logger: RazorLogger,
+    ) {}
+
+    public register() {
+        vscode.commands.registerCommand('razor/runCodeAction', (request: object) => this.runCodeAction(request), this);
+    }
+
+    private async runCodeAction(request: any): Promise<boolean> {
+        const response: CodeActionResolutionResponse = await this.serverClient.sendRequest('razor/resolveCodeAction', {Action: request.Action, Data: request.Data});
+        let workspaceEdit: vscode.WorkspaceEdit;
+        try {
```

Comment:
Should this `try` encompass the `sendRequest` above?

Reply:
It's left naked in other similar cases in the codebase I think. Plus shouldn't handling that kind of error occur somewhere higher up?

Reply:
It could, the log below just made me feel you had different ideas on what should be "caught". Also, I just noticed the `logAlways` below should probably be utilizing the `logError` method

Reply:
I switched it to logError in the latest patch :)

Reply:
I think it is our responsibility to validate the data we're getting within the request, but if the request itself is failing there's most likely an issue going on with the language server as a whole - to be dealt with at a higher level.